rpc/apis.go

```diff
@@ -35,94 +37,124 @@ const (
 					Public:    true,
 				},
-				rpc.API{
+				{
 					Namespace: EthNamespace,
 					Version:   apiVersion,
 					Service:   filters.NewPublicAPI(ctx.Logger, clientCtx, tmWSClient, evmBackend),
 					Public:    true,
 				},
-			)
-		case Web3Namespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		Web3Namespace: func(*server.Context, client.Context, *rpcclient.WSClient) []rpc.API {
+			return []rpc.API{
+				{
 					Namespace: Web3Namespace,
 					Version:   apiVersion,
 					Service:   web3.NewPublicAPI(),
 					Public:    true,
 				},
-			)
-		case NetNamespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		NetNamespace: func(_ *server.Context, clientCtx client.Context, _ *rpcclient.WSClient) []rpc.API {
+			return []rpc.API{
+				{
 					Namespace: NetNamespace,
 					Version:   apiVersion,
 					Service:   net.NewPublicAPI(clientCtx),
 					Public:    true,
 				},
-			)
-		case PersonalNamespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		PersonalNamespace: func(ctx *server.Context, clientCtx client.Context, _ *rpcclient.WSClient) []rpc.API {
+			evmBackend := backend.NewEVMBackend(ctx, ctx.Logger, clientCtx)
+			return []rpc.API{
+				{
 					Namespace: PersonalNamespace,
 					Version:   apiVersion,
 					Service:   personal.NewAPI(ctx.Logger, clientCtx, evmBackend),
 					Public:    false,
 				},
-			)
-		case TxPoolNamespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		TxPoolNamespace: func(ctx *server.Context, _ client.Context, _ *rpcclient.WSClient) []rpc.API {
+			return []rpc.API{
+				{
 					Namespace: TxPoolNamespace,
 					Version:   apiVersion,
 					Service:   txpool.NewPublicAPI(ctx.Logger),
 					Public:    true,
 				},
-			)
-		case DebugNamespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		DebugNamespace: func(ctx *server.Context, clientCtx client.Context, _ *rpcclient.WSClient) []rpc.API {
+			evmBackend := backend.NewEVMBackend(ctx, ctx.Logger, clientCtx)
+			return []rpc.API{
+				{
 					Namespace: DebugNamespace,
 					Version:   apiVersion,
 					Service:   debug.NewAPI(ctx, evmBackend, clientCtx),
 					Public:    true,
 				},
-			)
-		case MinerNamespace:
-			apis = append(apis,
-				rpc.API{
+			}
+		},
+		MinerNamespace: func(ctx *server.Context, clientCtx client.Context, _ *rpcclient.WSClient) []rpc.API {
+			evmBackend := backend.NewEVMBackend(ctx, ctx.Logger, clientCtx)
+			return []rpc.API{
+				{
 					Namespace: MinerNamespace,
 					Version:   apiVersion,
 					Service:   miner.NewPrivateAPI(ctx, clientCtx, evmBackend),
 					Public:    false,
 				},
-			)
-		default:
-			ctx.Logger.Error("invalid namespace value", "namespace", selectedAPIs[index])
+			}
+		},
+	}
+}
+
+// GetRPCAPIs returns the list of all APIs
+func GetRPCAPIs(ctx *server.Context, clientCtx client.Context, tmWSClient *rpcclient.WSClient, selectedAPIs []string) []rpc.API {
+	var apis []rpc.API
+	// remove duplicates
+	selectedAPIs = unique(selectedAPIs)
```

Comment:
we can remove the `unique` function, it should be checked by the config validation IIRC.

```suggestion
```